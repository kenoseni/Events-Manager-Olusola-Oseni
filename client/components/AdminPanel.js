import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import AdminPanelTable from './AdminPanelTable';
import Pages from './Pages';

class AdminPanel extends Component {
  render() {
    const { users, match, history, count, limit, userDetails } = this.props;
    return (
      <div className="container">
        <div className="row" id="">
          <div className="col-xs-12 col-md-0" />
          <div className="col-xs-12 col-md-12" id="etable">
            <div className="card w-40">
              <div className="card-body">
                <div className="table-wrapper">
                  <h3 id="appusers">Users</h3>
                  {/*<!--Table-->*/}
                  <table className="table table-hover table-responsive">
                    <thead className="blue-grey text-white lighten-1">
                      <tr>
                        <th>#</th>
                        <th className="th-lg">ID</th>
                        <th className="th-lg">FIRST NAME</th>
                        <th className="th-lg">LAST NAME</th>
                        <th className="th-lg">ISADMIN</th>
                        <th className="th-lg">ROLE</th>
                        <th className="th-lg">EMAIL</th>
                        {userDetails.userrole === 'superadmin' && (
                          <th className="th-lg">ACTION</th>
                        )}
                        {userDetails.userrole === 'admin' && (
                          <th className="th-lg" />
                        )}
                      </tr>
                    </thead>
                    <tbody id="adminpaneltable">
                      {users
                        ? users.map((user, i) => (
                            <AdminPanelTable
                              {...this.props}
                              key={i}
                              i={i + 1}
                              user={user}
                            />
                          ))
                        : null}
                    </tbody>
                  </table>
                  {/*<!--Table-->*/}
                  <div>
                    <div
                      style={{
                        display: 'flex',
                        justifyContent: 'space-between'
                      }}
                    >
                      <span className="count">
                        {users
                          ? `1 - ${users.length} of ${users.length}`
                          : null}
                      </span>
                      <span>
                        {count !== undefined && (
                          <Pages
                            count={count}
                            history={history}
                            limit={limit}
                          />
                        )}
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div className="col-xs-12 col-md-0" />
        </div>
      </div>
    );
  }
}

export default AdminPanel;
