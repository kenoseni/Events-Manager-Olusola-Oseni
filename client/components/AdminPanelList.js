import React, {Component} from 'react';
import PropTypes from 'prop-types';
import AdminPanel from './AdminPanel';
import NavBar from './Navbar';

class AdminPanelList extends Component {
  render () {
    const { users, match, history, searchForCenters } = this.props;
    const { count, limit } = this.props.user;
    return (
      <div style={{paddingTop: '90px'}}>
        <NavBar home='Home' admin='Admin' centers='Centers' events='Events' page='Admin' logout='Log Out' match={match} searchForCenters={searchForCenters} />
        <div className="container">
          <div className="row" id="users">
            <AdminPanel users={users} match={match} history={history} count={count} limit={limit} {...this.props}/>  
          </div>
        </div>
      </div>
    )
  }
}
AdminPanelList.propTypes = {
  users: PropTypes.array,
  match: PropTypes.object.isRequired
}
export default AdminPanelList
