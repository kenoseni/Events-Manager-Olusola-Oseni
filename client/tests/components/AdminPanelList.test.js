import React from 'react';
import { configure, shallow } from 'enzyme';
import sinon from 'sinon';
import Adapter from 'enzyme-adapter-react-16';
import Connected, { AdminPanels } from '../../components/AdminPanels';
import AdminPanelList from '../../components/AdminPanelList';
import AdminPanel from '../../components/AdminPanel';

configure({ adapter: new Adapter() });

describe('<AdminPanelList />', () => {
  let props = {
    user: {
      count: 0,
      limit: 2
    },
    users: [],
    isAuthenticated: true,
    isAdmin: true,
    userDetails: {
      count: 0
    },
    history: {
      location: {
        search: '/page=1'
      }
    },
    match: {}
  }
  const wrapper = shallow(<AdminPanelList
    {...props}/>
  )
  it('should render a <AdminPanelList /> component', () => {
    console.log(wrapper.debug())
    expect(wrapper).toMatchSnapshot();
  });
})
