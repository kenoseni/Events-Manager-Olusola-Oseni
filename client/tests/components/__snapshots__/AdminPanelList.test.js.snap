// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminPanelList /> should render a <AdminPanelList /> component 1`] = `
<div
  style={
    Object {
      "paddingTop": "90px",
    }
  }
>
  <Navbar
    admin="Admin"
    centers="Centers"
    events="Events"
    home="Home"
    logout="Log Out"
    match={Object {}}
    page="Admin"
  />
  <div
    className="container"
  >
    <div
      className="row"
      id="users"
    >
      <AdminPanel
        count={0}
        history={
          Object {
            "location": Object {
              "search": "/page=1",
            },
          }
        }
        isAdmin={true}
        isAuthenticated={true}
        limit={2}
        match={Object {}}
        user={
          Object {
            "count": 0,
            "limit": 2,
          }
        }
        userDetails={
          Object {
            "count": 0,
          }
        }
        users={Array []}
      />
    </div>
  </div>
</div>
`;
